### ISSUE TITLE
[DETECTION] DexProtector split apk is not detected (armeabi_v7a)

### ISSUE THREAD

**Issue by enovella** [rednaga/APKiD#303]
https://play.google.com/store/apps/details?id=com.napsternetlabs.napsternetv

```sh
[15:20 edu@xps tmp] >  extract split_config.armeabi_v7a.apk 
Archive:  split_config.armeabi_v7a.apk
  inflating: split_config.armeabi_v7a.apk.unpack/AndroidManifest.xml  
  inflating: split_config.armeabi_v7a.apk.unpack/lib/armeabi-v7a/libalice.so  
  inflating: split_config.armeabi_v7a.apk.unpack/lib/armeabi-v7a/libbarhopper_v3.so  
  inflating: split_config.armeabi_v7a.apk.unpack/lib/armeabi-v7a/libdexprotector.so  
  inflating: split_config.armeabi_v7a.apk.unpack/lib/armeabi-v7a/libdexprotector_h.so  
  inflating: split_config.armeabi_v7a.apk.unpack/lib/armeabi-v7a/libgojni.so  
  inflating: split_config.armeabi_v7a.apk.unpack/lib/armeabi-v7a/libimage_processing_util_jni.so  
  inflating: split_config.armeabi_v7a.apk.unpack/lib/armeabi-v7a/libpsitun2socks.so  
  inflating: split_config.armeabi_v7a.apk.unpack/lib/armeabi-v7a/libv2tun2socks.so  
  inflating: split_config.armeabi_v7a.apk.unpack/stamp-cert-sha256  
  inflating: split_config.armeabi_v7a.apk.unpack/META-INF/BNDLTOOL.SF  
  inflating: split_config.armeabi_v7a.apk.unpack/META-INF/BNDLTOOL.RSA  
  inflating: split_config.armeabi_v7a.apk.unpack/META-INF/MANIFEST.MF  

[15:20 edu@xps tmp] >  apkid split_config.armeabi_v7a.apk 
[+] APKiD 2.1.3 :: from RedNaga :: rednaga.io
```

```sh
[15:24 edu@xps armeabi-v7a] >  r2 libdexprotector.so 
Warning: run r2 with -e bin.cache=true to fix relocations in disassembly
 -- Disassemble?! No Disassemble Johnny No. 5!!!
[0x00000000]> x
- offset -   0 1  2 3  4 5  6 7  8 9  A B  C D  E F  0123456789ABCDEF
0x00000000  7f45 4c46 0101 0100 4450 4c46 0000 0000  .ELF....DPLF....
0x00000010  0300 2800 0100 0000 f84f 0000 3400 0000  ..(......O..4...
0x00000020  fcd7 0400 0002 0005 3400 2000 0900 2800  ........4. ...(.
0x00000030  1c00 1b00 0600 0000 3400 0000 3400 0000  ........4...4...
0x00000040  3400 0000 2001 0000 2001 0000 0400 0000  4... ... .......
0x00000050  0400 0000 0100 0000 0000 0000 0000 0000  ................
0x00000060  0000 0000 40d3 0000 40d3 0000 0500 0000  ....@...@.......
0x00000070  0010 0000 0100 0000 40d3 0000 40e3 0000  ........@...@...
0x00000080  40e3 0000 3402 0000 3402 0000 0600 0000  @...4...4.......
0x00000090  0010 0000 0100 0000 74d5 0000 74f5 0000  ........t...t...
0x000000a0  74f5 0000 3100 0400 5c00 0400 0600 0000  t...1...\.......
0x000000b0  0010 0000 0200 0000 60d3 0000 60e3 0000  ........`...`...
0x000000c0  60e3 0000 0001 0000 0001 0000 0600 0000  `...............
0x000000d0  0400 0000 52e5 7464 40d3 0000 40e3 0000  ....R.td@...@...
0x000000e0  40e3 0000 3402 0000 c00c 0000 0400 0000  @...4...........
0x000000f0  0100 0000 51e5 7464 0000 0000 0000 0000  ....Q.td........
[0x00000000]> q

[15:24 edu@xps armeabi-v7a] >  apkid libdexprotector.so 
[+] APKiD 2.1.3 :: from RedNaga :: rednaga.io
```

**Comment by apkunpacker** [comment]
> https://play.google.com/store/apps/details?id=com.napsternetlabs.napsternetv
> 
> ```shell
> [15:20 edu@xps tmp] >  extract split_config.armeabi_v7a.apk 
> Archive:  split_config.armeabi_v7a.apk
>   inflating: split_config.armeabi_v7a.apk.unpack/AndroidManifest.xml  
>   inflating: split_config.armeabi_v7a.apk.unpack/lib/armeabi-v7a/libalice.so  
>   inflating: split_config.armeabi_v7a.apk.unpack/lib/armeabi-v7a/libbarhopper_v3.so  
>   inflating: split_config.armeabi_v7a.apk.unpack/lib/armeabi-v7a/libdexprotector.so  
>   inflating: split_config.armeabi_v7a.apk.unpack/lib/armeabi-v7a/libdexprotector_h.so  
>   inflating: split_config.armeabi_v7a.apk.unpack/lib/armeabi-v7a/libgojni.so  
>   inflating: split_config.armeabi_v7a.apk.unpack/lib/armeabi-v7a/libimage_processing_util_jni.so  
>   inflating: split_config.armeabi_v7a.apk.unpack/lib/armeabi-v7a/libpsitun2socks.so  
>   inflating: split_config.armeabi_v7a.apk.unpack/lib/armeabi-v7a/libv2tun2socks.so  
>   inflating: split_config.armeabi_v7a.apk.unpack/stamp-cert-sha256  
>   inflating: split_config.armeabi_v7a.apk.unpack/META-INF/BNDLTOOL.SF  
>   inflating: split_config.armeabi_v7a.apk.unpack/META-INF/BNDLTOOL.RSA  
>   inflating: split_config.armeabi_v7a.apk.unpack/META-INF/MANIFEST.MF  
> 
> [15:20 edu@xps tmp] >  apkid split_config.armeabi_v7a.apk 
> [+] APKiD 2.1.3 :: from RedNaga :: rednaga.io
> ```

aarch64 is working fine
```sh
$ apkid 'NapsternetV_31.3.0.apks'
[+] APKiD 2.1.3 :: from RedNaga :: rednaga.io
[*] NapsternetV_31.3.0.apks!base.apk
 |-> packer : DexProtector
[*] NapsternetV_31.3.0.apks!base.apk!classes.dex
 |-> anti_vm : Build.DEVICE check, Build.FINGERPRINT check, Build.MANUFACTURER check
 |-> compiler : dx
[*] NapsternetV_31.3.0.apks!base.apk!classes2.dex
 |-> compiler : dx
[*] NapsternetV_31.3.0.apks!base.apk!classes3.dex
 |-> compiler : dx
[*] NapsternetV_31.3.0.apks!base.apk!classes4.dex
 |-> anti_vm : Build.FINGERPRINT check, Build.MANUFACTURER check
 |-> compiler : dx
[*] NapsternetV_31.3.0.apks!base.apk!classes5.dex
 |-> anti_debug : Debug.isDebuggerConnected() check
 |-> anti_vm : Build.FINGERPRINT check, Build.HARDWARE check, Build.MANUFACTURER check, Build.MODEL check, Build.PRODUCT check, Build.TAGS check, possible VM check
 |-> compiler : dx
[*] NapsternetV_31.3.0.apks!base.apk!classes6.dex
 |-> anti_vm : Build.MANUFACTURER check, Build.MODEL check, Build.TAGS check
 |-> compiler : dx
[*] NapsternetV_31.3.0.apks!base.apk!classes7.dex
 |-> compiler : dx
[*] NapsternetV_31.3.0.apks!split_config.arm64_v8a.apk!lib/arm64-v8a/libdexprotector.so
 |-> obfuscator : DexProtector
[*] NapsternetV_31.3.0.apks!split_config.arm64_v8a.apk!lib/arm64-v8a/libdexprotector_h.so
 |-> obfuscator : DexProtector
```

**Comment by apkunpacker** [comment]
> This is not matching properly https://github.com/rednaga/APKiD/blob/master/apkid/rules/elf/obfuscators.yara#L658

yeah for armv7 , this rule doesn't hit

```sh
$ apkid 'com.napsternetlabs.napsternetv.apks'
[+] APKiD 2.1.3 :: from RedNaga :: rednaga.io
[*] com.napsternetlabs.napsternetv.apks
 |-> packer : DexProtector
[*] com.napsternetlabs.napsternetv.apks!base.apk
 |-> packer : DexProtector
[*] com.napsternetlabs.napsternetv.apks!base.apk!classes.dex
 |-> anti_vm : Build.DEVICE check, Build.FINGERPRINT check, Build.MANUFACTURER check                                                 |-> compiler : dx
[*] com.napsternetlabs.napsternetv.apks!base.apk!classes2.dex
 |-> compiler : dx
[*] com.napsternetlabs.napsternetv.apks!base.apk!classes3.dex
 |-> compiler : dx
[*] com.napsternetlabs.napsternetv.apks!base.apk!classes4.dex
 |-> anti_vm : Build.FINGERPRINT check, Build.MANUFACTURER check
 |-> compiler : dx
[*] com.napsternetlabs.napsternetv.apks!base.apk!classes5.dex
 |-> anti_debug : Debug.isDebuggerConnected() check
 |-> anti_vm : Build.FINGERPRINT check, Build.HARDWARE check, Build.MANUFACTURER check, Build.MODEL check, Build.PRODUCT check, Build.TAGS check, possible VM check                                    |-> compiler : dx
[*] com.napsternetlabs.napsternetv.apks!base.apk!classes6.dex
 |-> anti_vm : Build.MANUFACTURER check, Build.MODEL check, Build.TAGS check                                                         |-> compiler : dx
[*] com.napsternetlabs.napsternetv.apks!base.apk!classes7.dex
 |-> compiler : dx
```

**Comment by apkunpacker** [comment]
Uploaded on VT in case file is missed
Armabi_v7a.apk
https://virustotal.com/gui/file/ed2486674e1cf1dcd9ad7fc17a5c0d50c1790071227ae236c976a1c92386ccff

Arm64.apks
https://virustotal.com/gui/file/b405d174561218eb6448eb18549db32625597ee72baff34f8153a7e32b7a019d

**Comment by enovella** [comment]
Fixed in #304 
```sh
[19:11 edu@xps]  (master)>  docker/apkid.sh /tmp/split_config.armeabi_v7a.apk
[+] APKiD 2.1.3 :: from RedNaga :: rednaga.io
[*] /input/split_config.armeabi_v7a.apk!lib/armeabi-v7a/libdexprotector.so
 |-> obfuscator : DexProtector
[*] /input/split_config.armeabi_v7a.apk!lib/armeabi-v7a/libdexprotector_h.so
 |-> obfuscator : DexProtector
```
